### PR TITLE
Add resource naming instrumentation for jax-rs

### PR DESCRIPTION
**Pull request by tylerbenson**
This should improve naming for dropwizard and anything else using jax-rs:
* jersey
* cxf
* restlet

NOTE: This integration is currently off by default.  Enable it by setting the following system property:
`-Ddd.integration.jax-rs.enabled=true`

Or the following Environment Variable:
`DD_INTEGRATION_JAX_RS_ENABLED=true`

### CODE REVIEW

File: dd-java-agent/testing/src/main/java/datadog/trace/agent/test/AgentTestRunner.java [hunk 1]
```diff
@@ -75,18 +91,68 @@ public static synchronized void agentSetup() {
...
   @Before
   public void beforeTest() {
     TEST_WRITER.start();
+    INSTRUMENTATION_ERROR_COUNT.set(0);
+    assert TEST_TRACER.activeSpan() == null;
+  }
+
+  @After
+  public void afterTest() {
+    assert INSTRUMENTATION_ERROR_COUNT.get() == 0;
```
Comment (realark) [inline dd-java-agent/testing/src/main/java/datadog/trace/agent/test/AgentTestRunner.java]: Nice!

**Comment by avram** [comment]
Is there anything that would need to be done to ensure that this covers JBoss/RestEASY?

**Comment by tylerbenson** [comment]
@avram if there is an embedable version you can submit a PR with a test for it, otherwise manual testing. If you do see any problems with it please let me know.

**Comment by avram** [comment]
I identified some issues with Resteasy, and opened #234 to resolve them

**Pull request merged**